xonsh/tools.py

```diff
@@ -830,12 +830,13 @@ def command_not_found(cmd):
     """Uses the debian/ubuntu command-not-found utility to suggest packages for a
     command that cannot currently be found.
     """
-    import shutil
-
     if not ON_LINUX:
         return ""
 
-    cnf = shutil.which("command-not-found")
+    cnf = builtins.__xonsh__.commands_cache.lazyget(
+        "command-not-found", ("/usr/lib/command-not-found",)
+    )[0]
```

Comment:
I think this needs to be a little more complex, sorry.  We need to check that the binary actually exists. If it does not, than `cnf` needs to be set to None so that the following block works. This is espcially true when `/usr/lib/command-not-found` is being used.

Reply:
Bah, I have tested two variants when command-not-found is in the path and under /usr/lib but totally missed when it is not available. Good catch!

Reply:
Done. I have put the original `isfile()` check back in. There is still no check if the file is executable but I think it is a weird enough corner case to not care about it (as the original code did not care about it either)

Reply:
haha no worries!